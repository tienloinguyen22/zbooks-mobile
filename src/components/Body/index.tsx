import { Body } from 'native-base';

export { Body };
