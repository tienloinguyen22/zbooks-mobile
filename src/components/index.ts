export * from './Body';
export * from './Button';
export * from './Card';
export * from './CardItem';
export * from './Container';
export * from './Content';
export * from './DatePicker';
export * from './ErrorBoundary';
export * from './ErrorText';
export * from './Field';
export * from './Footer';
export * from './Header';
export * from './Icon';
export * from './Image';
export * from './Input';
export * from './Item';
export * from './Label';
export * from './Left';
export * from './List';
export * from './ListItem';
export * from './Loading';
export * from './Picker';
export * from './Right';
export * from './ScrollView';
export * from './Text';
export * from './Touchable';
export * from './Thumbnail';
export * from './Tooltip';
export * from './View';
export * from './WithStore';
export * from './WithScreen';
export * from './Alert';
export * from './Animation';
export * from './AlertContainer';
