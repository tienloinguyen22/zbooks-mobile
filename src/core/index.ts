export * from './i18n';
export * from './themes/colors';
export * from './interfaces/ScreenProps';
export * from './interfaces/LoginType';
export * from './interfaces/User';
export * from './interfaces/Resource';
export * from './interfaces/Theme';
export * from './helpers/show_notification';
export * from './helpers/sleep';
export * from './helpers/record_error';
export * from './helpers/configure_google_sign_in';
export * from './helpers/combine_styles';
export * from './helpers/configure_google_sign_in';
export * from './helpers/handle_error';
export * from './helpers/get_primary_color';
export * from './helpers/format_date';
export * from './screen_names';
