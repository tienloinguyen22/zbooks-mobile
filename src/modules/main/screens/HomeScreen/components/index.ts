export * from './NavigationSample';
export * from './IconSample';
export * from './RematchSample';
export * from './NotificationSample';
export * from './CrashSample';
export * from './PickerSample';
export * from './AnalyticsSample';
export * from './TextStyleSample';
