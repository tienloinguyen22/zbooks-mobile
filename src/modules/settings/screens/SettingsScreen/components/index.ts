export * from './SettingMenuItem';
export * from './LogoutModal';
