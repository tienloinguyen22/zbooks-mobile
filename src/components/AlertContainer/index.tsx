import React from 'react';
import { ViewStyle } from 'react-native';
import Modal from 'react-native-modal';
import { combineStyles, colors, THEME_DARK } from '@app/core';
import { useTranslation } from 'react-i18next';
import { useTheme } from '@app/hooks';
import { Icon } from '../Icon';
import { Button } from '../Button';
import { View } from '../View';
import { styles } from './styles';
import { Text } from '../Text';

export type AlertType = 'SUCCESS' | 'WARNING' | 'ERROR' | 'INFO' | 'DEFAULT';

export interface AlertAction {
  title: string;
  onPress: () => void;
  outline?: boolean;
}

interface AlertProps {
  type: AlertType;
  title?: string;
  message: string;
  actions?: AlertAction[];
  onPressClose: () => void;
  closeable?: boolean;
}

export const AlertContainer = (props: AlertProps): JSX.Element => {
  const { primaryColor, theme } = useTheme();
  const { t } = useTranslation();

  const backgroundAlertStyle = {
    backgroundColor: theme === THEME_DARK ? colors.lightBlack : colors.white,
  };

  const messageTextStyle = {
    color: theme === THEME_DARK ? colors.white : colors.black,
  };

  const closeModal = (): void => {
    props.closeable !== false && props.onPressClose();
  };

  let titleContainerStyle = combineStyles<ViewStyle>(styles.titleDefault, {
    backgroundColor: primaryColor,
  });
  let title = t('dialog.info');
  let iconName = 'information-outline';

  switch (props.type) {
    case 'WARNING':
      titleContainerStyle = combineStyles<ViewStyle>(titleContainerStyle, {
        backgroundColor: colors.orange,
      });
      title = t('dialog.warning');
      iconName = 'alert-circle-outline';
      break;
    case 'ERROR':
      titleContainerStyle = combineStyles<ViewStyle>(titleContainerStyle, {
        backgroundColor: colors.red,
      });
      title = t('dialog.error');
      iconName = 'alert-outline';
      break;
    case 'SUCCESS':
      titleContainerStyle = combineStyles<ViewStyle>(titleContainerStyle, {
        backgroundColor: colors.green,
      });
      title = t('dialog.success');
      iconName = 'check';
      break;
    default:
      break;
  }

  props.title && ({ title } = props);

  return (
    <View style={styles.container}>
      <Modal isVisible={true} onBackButtonPress={closeModal} onBackdropPress={closeModal}>
        <View style={[styles.alertContainer, backgroundAlertStyle]}>
          <View style={titleContainerStyle}>
            <Icon name={iconName} color={colors.white} size={25} />
            <Text style={styles.titleText}>{title}</Text>
            <View style={styles.dialogIcon} />
          </View>
          <Text style={[styles.messageText, messageTextStyle]}>{props.message}</Text>
          {props.actions && (
            <View style={styles.buttonContainer}>
              {props.actions.map((item) => {
                return (
                  <Button outline={item.outline} onPress={item.onPress} key={item.title} style={styles.buttonDialog}>
                    <Text white={!item.outline} primary={item.outline}>
                      {item.title}
                    </Text>
                  </Button>
                );
              })}
            </View>
          )}
          {!props.actions && (
            <View style={styles.reserveContainer}>
              <Button onPress={props.onPressClose} style={styles.buttonDialog}>
                <Text white>{t('dialog.close')}</Text>
              </Button>
            </View>
          )}
        </View>
      </Modal>
    </View>
  );
};
