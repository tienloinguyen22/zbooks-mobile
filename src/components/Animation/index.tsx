import React from 'react';
import { Resource } from '@app/core';
import LottieView from 'lottie-react-native';

interface Props {
  width: number;
  height: number;
  source: Resource;
  autoPlay?: boolean;
  loop?: boolean;
}

export const Animation = (props: Props): JSX.Element => {
  return (
    <LottieView
      source={props.source}
      style={{
        width: props.width,
        height: props.height,
      }}
      autoPlay={props.autoPlay === true}
      loop={props.loop === true}
    />
  );
};
