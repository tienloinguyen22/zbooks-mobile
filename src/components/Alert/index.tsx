import React from 'react';
import RootSiblings from 'react-native-root-siblings';
import { AlertContainer, AlertAction, AlertType } from '../AlertContainer';

interface ShowParams {
  type: AlertType;
  title?: string;
  message: string;
  actions?: AlertAction[];
  onPressClose?: () => void;
  closeable?: boolean;
}

let alert: RootSiblings | undefined;
const hide = (): void => {
  alert && alert.destroy();
};

const show = (props: ShowParams): RootSiblings => {
  alert && hide();
  alert = new RootSiblings(<AlertContainer {...props} onPressClose={props.onPressClose || hide} />);
  return alert;
};

export const Alert = {
  show,
  hide,
};
