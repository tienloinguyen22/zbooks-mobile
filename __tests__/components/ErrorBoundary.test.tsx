import React from 'react';
import { render } from '@testing-library/react-native';
import { ErrorBoundary, Text } from '@app/components';
import { THEME_DARK, recordError } from '@app/core';
import { mockTheme } from '../test_helpers';

jest.mock('@app/core/helpers/record_error', () => ({
  recordError: jest.fn(),
}));

describe('components/ErrorBoundary', () => {
  beforeEach(() => {
    mockTheme(THEME_DARK);
  });

  it('renders successfully', async () => {
    const { baseElement } = render(
      <ErrorBoundary>
        <Text>Text</Text>
      </ErrorBoundary>,
    );

    expect(baseElement).toMatchSnapshot();
  });

  it('renders with errors successfully', async () => {
    ((console as unknown) as Console).error = jest.fn();

    const ErrorComponent = (): JSX.Element => <Text>{((undefined as unknown) as number).toString()}</Text>;
    const { baseElement } = render(
      <ErrorBoundary>
        <ErrorComponent />
      </ErrorBoundary>,
    );

    expect(((console as unknown) as Console).error).toBeCalledTimes(1);
    expect(recordError).toBeCalledTimes(1);
    expect(baseElement).toMatchSnapshot();
  });
});
