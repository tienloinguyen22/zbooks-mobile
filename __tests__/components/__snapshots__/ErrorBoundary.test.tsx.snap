// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ErrorBoundary renders successfully 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "textTransform": "none",
        }
      }
    >
      Text
    </Text>
  </View>
</View>
`;

exports[`components/ErrorBoundary renders with errors successfully 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FC4173",
          "fontSize": 13,
          "textTransform": "none",
        }
      }
    >
      error.somethingWentWrong
    </Text>
  </View>
</View>
`;
